hypothesis/lms
# Add annotation counts for users and groups by various facets

For:

 * https://github.com/hypothesis/report/issues/235

Based on:

 * https://github.com/hypothesis/lms/pull/5515

This PR:

 * Breaks a dependency between `users` ON `user_activity`
 * Moves the users before the activity counts
 * Adds:
   * `user annotation_counts`
   * `group_annotation_counts`
   * `group_bubbled_annotation_counts`

These new views are all variations of the `annotation_count` tables sliced up in various ways. These include most facets we have right now:

 * Time (week)
 * User
 * Group
 * Annotation sub-type
 * Sharing status

## Refactoring notes

There are a whole mess of very similar tables right now. I'm hoping to come around again and slim this down after we deliver the feature.

## Testing notes

 * Start all the things
 * Make some different annotations in LMS as different users (login as a teacher and a student)
 * Follow: https://stackoverflowteams.com/c/hypothesis/questions/513
 * You only need to progress as far as LMS
 * Use your DB viewer of choice to see the new tables
